Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -928,7 +928,7 @@ GCMemcardRemoveFileRetVal GCMemcard::RemoveFile(u8 index)  // index in the direc
   // here that has an empty file with the filename "Broken File000" where the actual deleted file
   // was. Determine when exactly this happens and if this is neccessary for anything.
 
-  memset(&(UpdatedDir.m_dir_entries[index]), 0xFF, DENTRY_SIZE);
+  memset(static_cast<void*>(&(UpdatedDir.m_dir_entries[index])), 0xFF, DENTRY_SIZE);
```

Comment:
This should probably not actually be silenced like this and instead have the initializtion and/or invalidation be done properly within a member function.

Reply:
> While I appreciate the gusto on trying to clean up some of these warnings, some of these changes seem to just silence warnings rather than properly fixing the underlying cause of said warnings. Particularly with ones that cast the destination type in memset calls to `void*`.
> 
> The commit messages also don't state what warnings were actually being resolved within them.

Your comments are fair. Let me split the patch and fix some places as you suggested.